### PR TITLE
[TECHNICAL-SUPPORT] LPS-65975

### PR DESCRIPTION
**Pull request by SamZiemer**
/cc @mbowerman

### CODE REVIEW

**Comment by SamZiemer** [comment]
ci:test

File: portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			runSQL(
+				StringBundler.concat(
+					"update LayoutSet set themeId = '", newThemeId,
+					"' where themeId = '", oldThemeId, "'"));
+		}
+	}
+
+	private static final String[][] _RENAMED_THEME_IDS = {
+		new String[] {"classic", "classic_WAR_classictheme"},
+		new String[] {"admin", "admin_WAR_admintheme"}
```
Comment (achaparro) [inline portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Hi @mbowerman, @SamZiemer,

The pull looks good but I have one question, which is the admin theme? where can I find it to deploy it in my tests?

Thanks.

Comment (SamZiemer) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Hi @achaparro, the admin theme can be found in modules/apps/foundation/frontend-theme/frontend-theme-admin

Comment (achaparro) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: @SamZiemer does it exist in previous versions of Liferay (6.2 and below)?

Thanks.

Comment (SamZiemer) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Hmm, I'm not sure where the admin theme is in 6.2 or below.

Comment (achaparro) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Then we don't need to upgrade it, right? admin can be a common word for naming a theme in custom developments so if we didn't use it in 6.2 and below we shouldn't upgrade it. Please, correct me if I am wrong.

Thanks.

Comment (mbowerman) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: It looks like it was called "controlpanel" in 6.2 and earlier. See https://github.com/liferay/liferay-portal-ee/commit/161b85f7823da9ab42bced1fcfb3f13a3c477ae7. So maybe we should add a third value to the array that maps "controlpanel" to "admin_WAR_admintheme"?

Comment (mbowerman) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: We would need to upgrade it for customers who were on a version of DXP that were on the base version of DXP (without LPS-65845). Actually, I'm not even sure if it's possible to select the Admin theme for a LayoutSet. So maybe we wouldn't need to upgrade it anyway.

Comment (achaparro) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Hi @mbowerman, thanks for the info. In that case I think that best option is to have just two options replacing "admin" for controlpanel.

is that ok for you?

Comment (mbowerman) [reply portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java]: Sorry, I don't understand your proposal. I think we should do this:
    
```
private static final String[][] _RENAMED_THEME_IDS = {
	new String[] {"classic", "classic_WAR_classictheme"},
	new String[] {"admin", "admin_WAR_admintheme"},
	new String[] {"controlpanel", "admin_WAR_admintheme"}
};
```

That way it upgrades properly for people who are coming from 6.2 or earlier, as well as for people who are coming from the base DXP installation. Is this what you had in mind?

**Comment by achaparro** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by achaparro** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/55810. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**